indico/modules/rb_new/client/js/actions.js

```diff
@@ -362,3 +370,36 @@ export function fetchTimelineData() {
         dispatch(updateTimelineData(response.data));
     };
 }
+
+export function createBooking(args) {
+    return async (dispatch) => {
+        let response;
+        const params = preProcessParameters(args, ajaxBookingRules);
+        dispatch({type: BOOKING_ONGOING});
+
+        try {
+            response = await indicoAxios.post(createBookingURL(), params);
+        } catch (error) {
+            handleAxiosError(error);
```

Comment:
Have a look at [createSubmissionError](https://github.com/indico/indico/pull/3391/files#diff-67b3b116508a9dc437d4535a38add0c7R22); like that you can mark form fields as erroneous based on the error from webargs. Probably needs some adjustments to use with an ajax request from a redux action

Reply:
Do I need that, though? I have already a client-side validator. If the user manages to mess it up, it's their fault that they'll see an error dialog.

Reply:
In that case probably not - but I think it'd be nice to catch such errors nicely. Could have a generic util that takes care of it at some point. Not super important right now though